### PR TITLE
Add multiselect list, sortable list, and status list design patterns

### PR DESCRIPTION
**Pull request by daileytj**
Fixes PXBLUE-1323, 1326, 1327.

#### Changes proposed in this Pull Request:
- Add multiselect list design pattern
- Add sortable list design pattern
- Add status list design pattern

### CODE REVIEW

**Comment by emclaug2** [comment]
I don't think the multi-select list should have the chevron enabled if the entire item isn't clickable.

**Comment by emclaug2** [comment]
Can you adjust the right-padding of the items in the sortable list?
image

**Comment by emclaug2** [comment]
Since you pulled these patterns over from the angular-design-pattern repo this all looks good to me, just see my two comments above.

**Review by emclaug2 — APPROVED** [review]

**Review by joebochill — COMMENTED** [review]
Functionally, everything looks good. 
- Bottomsheets need to avoid the unsafe area on mobile:
image
There should be some reference for doing this in the old ionic examples.


Following Evan's comment on sortable, there seem to be some icon alignment issues. These are all present in Angular as well. **Can we log a bug (one for angular one for ionic and cross-link them) and go in and crush these all later**:
image
image
image

These are mostly caused when we use an iconButton in place of an icon which causes extra spacing/size.

**Comment by joebochill** [comment]
One other thing we need to do is take out the menu icon logic in the toolbar. Since this is a 'mobile-only' example, we should always see that menu icon at all screen sizes.

File: src/app/pages/list/action-list/action-list.component.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <mat-toolbar data-cy="pxb-toolbar" color="primary" class="app-bar mat-elevation-z2">
-    <button data-cy="toolbar-menu" *ngIf="isSmall" style="margin-left: -8px;" mat-icon-button (click)="toggleMenu()">
+    <button data-cy="toolbar-menu" style="margin-left: -8px;" mat-icon-button (click)="toggleMenu()">
         <mat-icon>menu</mat-icon>
     </button>
-    <div class="mat-title" [style.marginLeft.px]="isSmall ? 24 : 0">Action List</div>
+    <div class="mat-title" style="margin-left: 24px;">Action List</div>
     <pxb-spacer></pxb-spacer>
     <button data-cy="toolbar-delete" mat-icon-button (click)="onRemoveAll()" style="margin-right: 8px;">
```
Comment (joebochill) [inline src/app/pages/list/action-list/action-list.component.html:7]: If there's a margin right: 8 on the button and a margin right: -8 on the icon, can we just remove them both?

File: src/app/pages/list/multiselect-list/multiselect-list.component.html [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
     </button>
 </mat-toolbar>
 <mat-list class="multi-select-list">
     <pxb-info-list-item data-cy="list-content" *ngFor="let item of data" [iconAlign]="'center'">
-        <mat-checkbox pxb-icon [checked]="isSelected(item)" (change)="onSelected(item)"></mat-checkbox>
+        <mat-checkbox
+            pxb-icon
+            [checked]="isSelected(item)"
+            (change)="onSelected(item)"
+            style="margin-left: -14px;"
```
Comment (joebochill) [inline src/app/pages/list/multiselect-list/multiselect-list.component.html:17]: would setting `iconAlign: left` instead of center do the same thing?

Comment (daileytj) [reply src/app/pages/list/multiselect-list/multiselect-list.component.html:17]: Tried it, this ends up being a little out of alignment too.
Screen Shot 2020-09-10 at 11 51 17 AM

File: src/app/pages/list/action-list/action-list.component.html (hunk 1 above)
Comment (joebochill) [reply src/app/pages/list/action-list/action-list.component.html:7]: image
image

Comment (joebochill) [reply src/app/pages/list/action-list/action-list.component.html:7]: nevermind...I'm a goof...there's 30 lines between these two things 🤦

**Review by joebochill — APPROVED** [review]

**Pull request merged**